src/widgets/table/table.rs

```diff
@@ -209,6 +209,24 @@ pub struct Table<'a> {
     segment_size: SegmentSize,
 }
 
+impl<'a> Default for Table<'a> {
+    fn default() -> Self {
+        Self {
+            rows: Default::default(),
+            header: Default::default(),
+            footer: Default::default(),
+            widths: Default::default(),
+            column_spacing: 1,
+            block: Default::default(),
+            style: Default::default(),
+            highlight_style: Default::default(),
+            highlight_symbol: Default::default(),
+            highlight_spacing: Default::default(),
+            segment_size: SegmentSize::None,
```

Comment:
Does this not work?

```suggestion
            segment_size: SegmentSize::None,
            column_spacing: 1,
            ..Default::default(),
```

Reply:
No - that recursively calls the method that we're defining. I tried the same thing ;)

Reply:
unfortunately this would require `Table` to have a `Default` in the first place :/

Reply:
It would be nice to do something like this:

```rust
#[derive(Default)]
struct Foo {
    alpha: u8,
    #[default = 1]
    beta: u8,
}
```

which is listed under future possibilities in this RFC:

https://rust-lang.github.io/rfcs/3107-derive-default-enum.html#overriding-default-fields